app/concerns/concealable.rb

```diff
@@ -7,13 +7,16 @@ module Concealable
       private: 1,
       access_list: 2,
       registered: 3,
+      legacy: 4,
     }, _prefix: true
 
     def visible_to?(user)
       # does not support access lists at this time
       return true if privacy_public?
       return false unless user
       return true if privacy_registered?
+      return false if user.read_only?
```

Comment:
Do we ever now, or plan to, use visible_to? on something that read-only users could own? (e.g. if we someday let them make indexes or something, iunno). read only cannot presently coexist with admin so that part at least seems fine.

Reply:
I could conceivably see letting them make indexes but I don't expect it? I think you'd just have to put the owner condition earlier if we did that, though.